pkg/resizer/csi_resizer.go

```diff
@@ -46,47 +41,49 @@ const (
 	resizerSecretNamespaceKey = "csi.storage.k8s.io/resizer-secret-namespace"
 )
 
-// NewCSIResizer creates a new resizer responsible for resizing CSI volumes.
-func NewCSIResizer(
+// NewResizer creates a new resizer responsible for resizing CSI volumes.
+func NewResizer(
 	address string, timeout time.Duration,
 	k8sClient kubernetes.Interface, informerFactory informers.SharedInformerFactory) (Resizer, error) {
-
-	conn, err := connection.Connect(address)
-	if err != nil {
-		return nil, fmt.Errorf("failed to connect to CSI driver: %v", err)
-	}
-
-	err = csirpc.ProbeForever(conn, timeout)
+	csiClient, err := csi.New(address, timeout)
 	if err != nil {
-		return nil, fmt.Errorf("failed probing CSI driver: %v", err)
+		return nil, err
 	}
 
-	name, err := csirpc.GetDriverName(context.Background(), conn)
+	driverName, err := getDriverName(csiClient, timeout)
 	if err != nil {
 		return nil, fmt.Errorf("get driver name failed: %v", err)
 	}
 
-	supports, err := supportsPluginControllerService(conn, timeout)
+	supportControllerService, err := supportsPluginControllerService(csiClient, timeout)
 	if err != nil {
 		return nil, fmt.Errorf("failed to check if plugin supports controller service: %v", err)
 	}
 
-	if !supports {
+	if !supportControllerService {
 		return nil, errors.New("CSI driver does not support controller service")
 	}
 
-	supports, err = supportsControllerResize(conn, timeout)
+	supportControllerResize, err := supportsControllerResize(csiClient, timeout)
 	if err != nil {
 		return nil, fmt.Errorf("failed to check if plugin supports controller resize: %v", err)
 	}
 
-	if !supports {
-		return nil, fmt.Errorf("CSI driver does not support controller resize")
+	if !supportControllerResize {
+		supportsNodeResize, err := supportsNodeResize(csiClient, timeout)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check if plugin supports node resize: %v", err)
+		}
+		if supportsNodeResize {
+			klog.Info("The CSI driver supports node resize only, using trivial resizer to handle resize requests")
+			return newTrivialResizer(driverName), nil
```

Comment:
is there a way to get some tests in? There isn't an e2e that tests this as well, so we have this somewhat untested.  We need an e2e or something that tests the scenario.

Reply:
OK, I will try to add some unit tests first.